### PR TITLE
Add GitHub issue and PR templates

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+**What this PR does / why we need it**:
+
+**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
+Fixes #
+
+**Special notes for your reviewer**:
+
+_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._
+
+**Release note**:
```
Comment (neolit123) [inline .github/PULL_REQUEST_TEMPLATE.md]: i don't think we have the release-note plugin enabled for this repo (i.e. `do-not-merge/release-note-label-needed` will not be applied to PRs without release notes).

also i guess a question is - do we need release notes @chuckha ?

Comment (neolit123) [reply .github/PULL_REQUEST_TEMPLATE.md]: hm, looks like this was added here already:
https://github.com/kubernetes-sigs/cluster-api-provider-docker/blob/master/.github/PULL_REQUEST_TEMPLATE.md

Comment (chuckha) [reply .github/PULL_REQUEST_TEMPLATE.md]: They are just kinda here by default, easy enough to ignore and there are even instructions for NONE so I think this is fine for consistency

**Pull request merged**